lib/Config.js

```diff
@@ -680,23 +681,27 @@ class Config extends EventEmitter {
         if (auth === 'file' || auth === 'mem' || auth === 'cdmi') {
             // Auth only checks for 'mem' since mem === file
             auth = 'mem';
-            let authfile = `${__dirname}/../conf/authdata.json`;
-            if (process.env.S3AUTH_CONFIG) {
-                authfile = process.env.S3AUTH_CONFIG;
-            }
-            let authData;
+            const authLoader = new AuthLoader();
             if (process.env.SCALITY_ACCESS_KEY_ID &&
-            process.env.SCALITY_SECRET_ACCESS_KEY) {
-                authData = buildAuthDataAccount(
-                  process.env.SCALITY_ACCESS_KEY_ID,
-                  process.env.SCALITY_SECRET_ACCESS_KEY);
+                process.env.SCALITY_SECRET_ACCESS_KEY) {
+                authLoader.addAccounts(buildAuthDataAccount(
+                    process.env.SCALITY_ACCESS_KEY_ID,
+                    process.env.SCALITY_SECRET_ACCESS_KEY));
             } else {
-                authData = require(authfile);
+                const authGlob = (process.env.S3AUTH_CONFIG ?
+                                  process.env.S3AUTH_CONFIG :
+                                  `${__dirname}/../conf/authdata.json`);
```

Comment:
just personal preference, instead of ternary, using an || statement to reduce some code